helm/cluster-vsphere/templates/machinedeployment.yaml

```diff
@@ -7,6 +7,8 @@ metadata:
   namespace: {{ $.Release.Namespace }}
   labels:
     {{- include "labels.common" $ | nindent 4 }}
+  annotations:
+    helm.sh/resource-policy: "keep"
```

Comment:
@fiunchinho @jkremser Did you test upgrades with config changes? We render MachineDeployment/MachinePool per nodePools and users can add/remove items from that array. In that case, we should delete undesired CRs.